test/test_sparqlupdatestore.py

```diff
@@ -88,12 +89,15 @@ def testSimpleGraph(self):
 
         g2.add((tarek, likes, pizza))
         g.remove((tarek, likes, pizza))
-        r = g.query("SELECT * WHERE { ?s <urn:likes> <urn:pizza> . }")
+        r = g.query("SELECT * WHERE { ?s <urn:example:likes> <urn:example:pizza> . }")
         self.assertEqual(1, len(list(r)), "only bob likes pizza")
 
     def testConjunctiveDefault(self):
         g = self.graph.get_context(graphuri)
         g.add((tarek, likes, pizza))
+
+        isunion = len(self.graph) > 0
```

Comment:
Won't this always be false? This changes the public API, is this intentional? If so it may be best to note it in the PR description.

Reply:
Aargh, that's overlooked detritus from not being able to replicate GH Action test environment, I've tried and failed to run two instances of fuseki a la `with_fuseki.sh` and didn't want to skip the tests, sigh. Reverted the `isunion` detritus.

Reply:
Not sure what OS you are on, but for me I run it like this and it works:

```bash
$ ./with-fuseki.sh .venv/bin/pytest test/test_issue1003.py --log-level DEBUG -rA
declare -- jena_uri="https://archive.apache.org/dist/jena/binaries/apache-jena-fuseki-3.17.0.tar.gz"
declare -- jena_archive="/home/iwana/.cache/apache-jena-fuseki-3.17.0.tar.gz"
declare -- XDG_CACHE_HOME="/home/iwana/.cache"
declare -- LOCALSTATEDIR="./var"
apache-jena-fuseki-3.17.0.tar.gz: OK
declare -- FUSEKI_HOME="./var/apache-jena-fuseki-3.17.0"
declare -- fuseki_base_normal="./var/apache-jena-fuseki-3.17.0/run-normal"
declare -- fuseki_base_tdb="./var/apache-jena-fuseki-3.17.0/run-tdb"
declare -- fuseki_pidfile_normal="./var/apache-jena-fuseki-3.17.0/run-normal/server.pid"
declare -- fuseki_pidfile_tdb="./var/apache-jena-fuseki-3.17.0/run-tdb/server.pid"
declare -- fuseki_port_normal="3030"
declare -- fuseki_port_tdb="3031"
declare -- fuseki_log_normal="./var/apache-jena-fuseki-3.17.0/run-normal/out.log"
declare -- fuseki_log_tdb="./var/apache-jena-fuseki-3.17.0/run-tdb/out.log"
INFO: starting fuseki: normal
INFO: starting fuseki: tdb
declare -- fuseki_pid_tdb="1077642"
declare -- fuseki_pid_normal="1077641"
INFO: checking if http://localhost:3030/ (fuseki-normal) is okay
INFO: url http://localhost:3030/ (fuseki-normal) not okay, waiting ...
INFO: url http://localhost:3030/ (fuseki-normal) is okay
INFO: checking if http://localhost:3031/ (fuseki-tdb) is okay
INFO: url http://localhost:3031/ (fuseki-tdb) is okay
INFO: pid 1077642 (fuseki-tdb) running
INFO: pid 1077641 (fuseki-normal) running
running: .venv/bin/pytest test/test_issue1003.py --log-level DEBUG -rA
============================================================================ test session starts ============================================================================
platform linux -- Python 3.8.12, pytest-6.2.5, py-1.11.0, pluggy-1.0.0
rootdir: /home/iwana/sw/d/github.com/iafork/rdflib, configfile: tox.ini
plugins: subtests-0.5.0, cov-3.0.0
collected 1 item                                                                                                                                                            

test/test_issue1003.py .                                                                                                                                              [100%]

================================================================================== PASSES ===================================================================================
______________________________________________________________________________ test_scenarios _______________________________________________________________________________
----------------------------------------------------------------------------- Captured log call -----------------------------------------------------------------------------
INFO     root:test_issue1003.py:111 trix = <?xml version="1.0" encoding="utf-8"?>
<TriX xml:base="http://two.org/"
  xmlns:brick="https://brickschema.org/schema/Brick#"
  xmlns:csvw="http://www.w3.org/ns/csvw#"
  xmlns:dc="http://purl.org/dc/elements/1.1/"
  xmlns:dcat="http://www.w3.org/ns/dcat#"
  xmlns:dcmitype="http://purl.org/dc/dcmitype/"
  xmlns:dcterms="http://purl.org/dc/terms/"
  xmlns:dcam="http://purl.org/dc/dcam/"
  xmlns:doap="http://usefulinc.com/ns/doap#"
  xmlns:foaf="http://xmlns.com/foaf/0.1/"
  xmlns:odrl="http://www.w3.org/ns/odrl/2/"
  xmlns:geo="http://www.opengis.net/ont/geosparql#"
  xmlns:org="http://www.w3.org/ns/org#"
  xmlns:owl="http://www.w3.org/2002/07/owl#"
  xmlns:prof="http://www.w3.org/ns/dx/prof/"
  xmlns:prov="http://www.w3.org/ns/prov#"
  xmlns:qb="http://purl.org/linked-data/cube#"
  xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#"
  xmlns:rdfs="http://www.w3.org/2000/01/rdf-schema#"
  xmlns:schema="https://schema.org/"
  xmlns:sh="http://www.w3.org/ns/shacl#"
  xmlns:skos="http://www.w3.org/2004/02/skos/core#"
  xmlns:sosa="http://www.w3.org/ns/sosa/"
  xmlns:ssn="http://www.w3.org/ns/ssn/"
  xmlns:time="http://www.w3.org/2006/time#"
  xmlns:vann="http://purl.org/vocab/vann/"
  xmlns:void="http://rdfs.org/ns/void#"
  xmlns:xsd="http://www.w3.org/2001/XMLSchema#"
  xmlns:xml="http://www.w3.org/XML/1998/namespace"
  xmlns:dct="http://purl.org/dc/terms/"
  xmlns="http://www.w3.org/2004/03/trix/trix-1/"
>
  <graph>
    <uri>http://g9.com/</uri>
    <triple>
      <uri></uri>
      <uri>http://www.w3.org/1999/02/22-rdf-syntax-ns#type</uri>
      <uri>http://www.w3.org/2004/02/skos/core#ConceptScheme</uri>
    </triple>
    <triple>
      <uri></uri>
      <uri>http://purl.org/dc/terms/source</uri>
      <uri>nick</uri>
    </triple>
    <triple>
      <uri></uri>
      <uri>http://purl.org/dc/terms/creator</uri>
      <uri>https://creator.com</uri>
    </triple>
  </graph>
  <graph>
    <uri>http://g8.com/</uri>
    <triple>
      <uri></uri>
      <uri>http://www.w3.org/1999/02/22-rdf-syntax-ns#type</uri>
      <uri>http://www.w3.org/2004/02/skos/core#ConceptScheme</uri>
    </triple>
    <triple>
      <uri></uri>
      <uri>http://purl.org/dc/terms/source</uri>
      <uri>nick</uri>
    </triple>
    <triple>
      <uri></uri>
      <uri>http://purl.org/dc/terms/creator</uri>
      <uri>https://creator.com</uri>
    </triple>
  </graph>
  <graph>
    <uri>urn:x-rdflib:default</uri>
  </graph>
</TriX>
========================================================================== short test summary info ==========================================================================
PASSED test/test_issue1003.py::test_scenarios
============================================================================= 1 passed in 0.03s =============================================================================
declare -- fuseki_pid_normal="1077641"
declare -- fuseki_pid_tdb="1077642"
INFO: killing pid 1077641 (fuseki-normal)
INFO: pid 1077641 (fuseki-normal) is not dead, waiting ...
INFO: pid 1077641 (fuseki-normal) is dead 
INFO: killing pid 1077642 (fuseki-tdb)
INFO: pid 1077642 (fuseki-tdb) is not dead, waiting ...
INFO: pid 1077642 (fuseki-tdb) is dead 
```

Please share the output if it fails, I can fix it.

Reply:
note I added some logging, should have no output on your branch.

Reply:
Thanks for that gem. Will make my life much easier.

Reply:
I'm seeing identical output.
However, judging by the above log and the assertions in the test:
```py
    trix = ds1.serialize(format="trix", base=Namespace("http://two.org/"))
    assert '<graph xml:base="http://one.org/">' in trix
    assert '<graph xml:base="http://two.org/">' in trix
    assert '<TriX xml:base="http://two.org/"' in trix
```
I can only see the last of the assertions as being true or is my understanding flawed?